ccl/backupccl: TestScheduleBackup failed

ccl/backupccl.TestScheduleBackup [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9066252?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9066252?buildTab=artifacts#/) on master @ [a36d88bebd1d26161c3c7327b86af72fca88fc2c](https://github.com/cockroachdb/cockroach/commits/a36d88bebd1d26161c3c7327b86af72fca88fc2c):


```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x39f29ec]

goroutine 114026 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x400231e6a8?, {0x68673f0, 0x40175c1380})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x68
panic({0x4a64880, 0x96967d0})
	GOROOT/src/runtime/panic.go:884 +0x20c
github.com/cockroachdb/cockroach/pkg/jobs.(*ScheduledJob).ExecutorType(...)
	github.com/cockroachdb/cockroach/pkg/jobs/scheduled_job.go:173
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.processSchedulePTSRecord({0x68673f0?, 0x4017f276b0?}, 0x400231e968?, {{0x40091a0ba0, 0x10, 0x10}, {0x174c668d633f8ee0, 0x0, 0x0}, 0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:227 +0x9c
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.manageProtectedTimestamps.func1({0x68673f0, 0x4017f276b0}, {0x68be080, 0x40049e6820})
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:124 +0x268
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn.func1({0x68673f0?, 0x4017f276b0?}, 0x68cdf60?)
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1406 +0x38
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn.func4({0x68673f0, 0x4017f276b0}, 0x4007be4790)
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1484 +0x2ac
github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1({0x68673f0?, 0x4017f276b0?}, 0x400231edb8?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:965 +0x2c
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0x4007be4790, {0x68673f0, 0x4017f276b0}, 0x400231f0e8)
	github.com/cockroachdb/cockroach/pkg/kv/txn.go:928 +0x70
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0x68673f0, 0x4017f276b0}, 0x4017f276b0?, 0x45f4ec?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:964 +0x58
github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl(0xffff85e79688?, {0x68673f0, 0x4017f276b0}, 0xf6d5f40?, 0x40?, 0x0?, 0x0?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:927 +0xa8
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(0x38?, {0x68673f0?, 0x4017f276b0?}, 0x0?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:902 +0x34
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn(0x4007dc9e60, {0x68673f0, 0x4017f276b0}, 0x4008915cf0, {0x0, 0x0, 0x9e01?})
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1472 +0x20c
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn(0x4a5e760?, {0x68673f0, 0x4017f276b0}, 0x401156da20, {0x0, 0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1407 +0x94
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.manageProtectedTimestamps({0x68673f0, 0x4017f276b0}, {0x68b94f0, 0x401bdac168})
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:105 +0xec
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.(*metricsPoller).Resume.func1({0x52df362?, 0x400231f448?}, 0x5688f08)
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/poller.go:64 +0xb8
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.(*metricsPoller).Resume(0x40126c5040?, {0x68673f0, 0x4017f276b0}, {0x4ff94e0?, 0x401bdac168})
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/poller.go:75 +0x238
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0x40126c5040?, 0x0?, {0x683a7d8?, 0x401bdac150?}, {0x68673f0?, 0x4012225260?}, {0x4ff94e0?, 0x401bdac168?}, 0x400231f860)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1529 +0xf4
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0x4003bd0d80, {0x68673f0, 0x40175c1650}, {0x4ff94e0, 0x401bdac168}, {0x683a7d8, 0x401bdac150}, 0x40126c5040, {0x400b7589f8, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1530 +0x6a0
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0x4003bd0d80, {0x68673f0, 0x40175c12f0}, {0x683a7d8, 0x401bdac150}, 0x40126c5040, {0x400b7589f8, 0x7}, {0x400b758bb0, 0x7})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:417 +0x500
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x4008940c00?, 0x4c09f4?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:335 +0xc0
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x128
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x358
=== RUN   TestScheduleBackup/cluster-backup_chaining=true
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScheduleBackup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25374

ccl/backupccl.TestScheduleBackup [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9070849?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9070849?buildTab=artifacts#/) on master @ [9b43dc50d468171bc9919dc684b7274b63e13da2](https://github.com/cockroachdb/cockroach/commits/9b43dc50d468171bc9919dc684b7274b63e13da2):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x78 pc=0x40d18bb]

```
Stack: 

```
goroutine 207982372 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x1e9392a?, {0x70b41b0, 0xc010583530})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x52b4260, 0x9f877d0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/jobs.(*ScheduledJob).ExecutorType(...)
	github.com/cockroachdb/cockroach/pkg/jobs/scheduled_job.go:173
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.processSchedulePTSRecord({0x70b41b0?, 0xc01ae6c390?}, 0xc02c4429a8?, {{0xc00f4660a0, 0x10, 0x10}, {0x174c8326610cc880, 0x0, 0x0}, 0x0, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:227 +0x9b
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.manageProtectedTimestamps.func1({0x70b41b0, 0xc01ae6c390}, {0x710aba0, 0xc01159cb40})
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:124 +0x345
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn.func1({0x70b41b0?, 0xc01ae6c390?}, 0xc02c442c30?)
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1406 +0x31
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn.func4({0x70b41b0, 0xc01ae6c390}, 0xc022b12210)
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1484 +0x3b4
github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1({0x70b41b0?, 0xc01ae6c390?}, 0xc008b36f00?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:965 +0x27
github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec(0xc022b12210, {0x70b41b0, 0xc01ae6c390}, 0xc02c443100)
	github.com/cockroachdb/cockroach/pkg/kv/txn.go:928 +0xae
github.com/cockroachdb/cockroach/pkg/kv.runTxn({0x70b41b0, 0xc01ae6c390}, 0xc01ae6c390?, 0x3b9aca00?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:964 +0x6b
github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl(0x7fb163556108?, {0x70b41b0, 0xc01ae6c390}, 0x2d0e3300?, 0xc0?, 0x0?, 0xc0089bfba0?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:927 +0xa7
github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn(0x463207?, {0x70b41b0?, 0xc01ae6c390?}, 0xc02c4432c8?)
	github.com/cockroachdb/cockroach/pkg/kv/db.go:902 +0x2d
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn(0xc012ee0b40, {0x70b41b0, 0xc01ae6c390}, 0xc01cf9eca0, {0x0, 0x0, 0x20?})
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1472 +0x2a7
github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn(0x52ae140?, {0x70b41b0, 0xc01ae6c390}, 0xc010448640, {0x0, 0x0, 0x0})
	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1407 +0x9f
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.manageProtectedTimestamps({0x70b41b0, 0xc01ae6c390}, {0x7105f90, 0xc01ba080f0})
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:105 +0xf0
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.(*metricsPoller).Resume.func1({0x5b2bb96?, 0xc02c443450?}, 0x5ed5668)
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/poller.go:64 +0xb1
github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.(*metricsPoller).Resume(0xc028f2a9c0?, {0x70b41b0, 0xc01ae6c390}, {0x5846ca0?, 0xc01ba080f0})
	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/poller.go:75 +0x31a
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc028f2a9c0?, 0x0?, {0x70877c8?, 0xc01ba080d8?}, {0x70b41b0?, 0xc03811d5f0?}, {0x5846ca0?, 0xc01ba080f0?}, 0xc02c443860)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1529 +0xcc
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc021160000, {0x70b41b0, 0xc010583d70}, {0x5846ca0, 0xc01ba080f0}, {0x70877c8, 0xc01ba080d8}, 0xc028f2a9c0, {0xc008114ee8, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1530 +0x8ed
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc021160000, {0x70b41b0, 0xc0105834a0}, {0x70877c8, 0xc01ba080d8}, 0xc028f2a9c0, {0xc008114ee8, 0x7}, {0xc00ef2cb88, 0x7})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:417 +0x5a6
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0xc029a08400?, 0xc005d22fd0?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:335 +0xfa
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470 +0x146
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:461 +0x43b
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/panic.go:884
*   | runtime.panicmem
*   | 	GOROOT/src/runtime/panic.go:260
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:835
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*ScheduledJob).ExecutorType
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/scheduled_job.go:173
*   | github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.processSchedulePTSRecord
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:227
*   | github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.manageProtectedTimestamps.func1
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:124
*   | github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1406
*   | github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn.func4
*   | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1484
*   | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:965
*   | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
*   | 	github.com/cockroachdb/cockroach/pkg/kv/txn.go:928
*   | github.com/cockroachdb/cockroach/pkg/kv.runTxn
*   | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:964
*   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl
*   | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:927
*   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
*   | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:902
*   | github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).txn
*   | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1472
*   | github.com/cockroachdb/cockroach/pkg/sql.(*InternalDB).Txn
*   | 	github.com/cockroachdb/cockroach/pkg/sql/internal.go:1407
*   | github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.manageProtectedTimestamps
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/job_statistics.go:105
*   | github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.(*metricsPoller).Resume.func1
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/poller.go:64
*   | github.com/cockroachdb/cockroach/pkg/jobs/metricspoller.(*metricsPoller).Resume
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/metricspoller/poller.go:75
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1529
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1530
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:417
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:335
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:470
*   | runtime.goexit
*   | 	GOROOT/src/runtime/asm_amd64.s:1594
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestScheduleBackup.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/jobs/metricspoller/job_statistics.go
+++ b/pkg/jobs/metricspoller/job_statistics.go
@@ -221,8 +221,11 @@ func processSchedulePTSRecord(
 ) error {
 	sj, err := jobs.ScheduledJobTxn(txn).
 		Load(ctx, scheduledjobs.ProdJobSchedulerEnv, scheduleID)
-	if jobs.HasScheduledJobNotFoundError(err) {
-		return nil // nolint:returnerrcheck -- schedule maybe deleted when we run; just keep going.
+	if err != nil {
+		if jobs.HasScheduledJobNotFoundError(err) {
+			return nil // nolint:returnerrcheck -- schedule maybe deleted when we run; just keep going.
+		}
+		return err
 	}
 	ex, err := jobs.GetScheduledJobExecutor(sj.ExecutorType())
 	if err != nil {

